### PR TITLE
judge updates

### LINKED ISSUE #5: How do I run this locally?
I don't see instructions

### LINKED ISSUE #8: Automatically Detect Trailing Whitespace on Sample Output
https://usaco.guide/general/io?lang=cpp#usaco-note---extra-whitespace

so you should get some warning when the following appears:

Screen Shot 2021-06-20 at 9 38 37 AM

### LINKED ISSUE #10: USACO Problem URLs
- [ ] Validate URL (before submit is pressed?)
- [ ] allow pasting of the just the ID (1137)
- [ ] Display the name of the problem somewhere, as well as the input / output format.

Screen Shot 2021-06-20 at 9 41 37 AM

### PR DESCRIPTION
**Pull request by bqi343**
- Closes #5
- Closes #8
- Addresses first two parts of #10. 
- ~~Disable "Reset File to Template" button for "view only" users~~
  - ~~TODO: update button styling~~
  - removed button for "view only" users
- ~~TODO: Install dependencies for third part of #10 (whoops idk how to do this).~~ modified backend, closes #10

### CODE REVIEW

File: src/components/JudgeInterface/JudgeInterface.tsx [hunk 1]
```diff
@@ -1,32 +1,57 @@
...
-    if (!problemID) {
-      alert('Error: Failed to parse problem ID. Is the usaco.org URL correct?');
-      return;
-    }
-
+    // if (!props.problemID) {
+    //   alert('Error: Failed to parse problem ID. Is the usaco.org URL correct?');
+    //   return;
+    // }
+    // ^ this should never happen
```
Comment (thecodingwizard) [inline src/components/JudgeInterface/JudgeInterface.tsx:54]: maybe just delete this?

File: src/components/JudgeInterface/JudgeInterface.tsx [hunk 2]
```diff
@@ -66,21 +91,67 @@ export default function JudgeInterface(): JSX.Element {
...
+                ? 'Could not parse problem data.'
+                : 'Problem ID does not exist.'}
+            </p>
+          </>
+        )}
         <p className="text-gray-400 text-sm">
           Early access. Report issues to Github. Do not spam submit.
-          {/* USACO 2020 US Open Contest, Bronze */}
+          {/* Note: You will not be able to submit to a problem in an active contest. */}
+          {/* ^ is this necessary? */}
```
Comment (thecodingwizard) [inline src/components/JudgeInterface/JudgeInterface.tsx:146]: idk I feel like some complete beginners might get confused and try to use this to submit :P

File: src/components/JudgeInterface/JudgeInterface.tsx [hunk 3]
```diff
@@ -151,7 +146,7 @@ export default function JudgeInterface(props: {
       </div>
       <button
         className="block w-full py-2 text-sm uppercase font-bold text-indigo-300 hover:text-indigo-100 bg-indigo-900 bg-opacity-50 focus:outline-none disabled:cursor-not-allowed"
-        disabled={!statusData || statusData.statusCode <= -8}
+        disabled={statusData?.statusCode ? statusData?.statusCode <= -8 : false}
```
Comment (thecodingwizard) [inline src/components/JudgeInterface/JudgeInterface.tsx:149]: maybe just `statusData?.statusCode && statusData.statusCode <= -8`?

Comment (bqi343) [reply src/components/JudgeInterface/JudgeInterface.tsx:149]: Screen Shot 2021-06-20 at 9 18 31 PM

Comment (thecodingwizard) [reply src/components/JudgeInterface/JudgeInterface.tsx:149]: hmm what about `!!statusData?.statusCode && statusData.statusCode <= -8`

Comment (thecodingwizard) [reply src/components/JudgeInterface/JudgeInterface.tsx:149]: Or maybe just `(statusData?.statusCode ?? 0) <= -8`

**Review by thecodingwizard — APPROVED** [review]

**Pull request merged**